base/util_testing.go

```diff
@@ -173,13 +172,28 @@ func TestUseXattrs() bool {
 	useXattrs := os.Getenv(TestEnvSyncGatewayUseXattrs)
 	switch {
 	case strings.ToLower(useXattrs) == strings.ToLower(TestEnvSyncGatewayTrue):
-		Infof(KeyAll, "Using xattrs: strings.ToLower(useXattrs) == strings.ToLower(TestEnvSyncGatewayTrue).  |%v| == |%v|", strings.ToLower(useXattrs), strings.ToLower(TestEnvSyncGatewayTrue))
 		return true
 	default:
-		Infof(KeyAll, "NOT Using xattrs: strings.ToLower(useXattrs) != strings.ToLower(TestEnvSyncGatewayTrue).  |%v| != |%v|", strings.ToLower(useXattrs), strings.ToLower(TestEnvSyncGatewayTrue))
+		// Otherwise fallback to hardcoded default
+		return DefaultUseXattrs
 	}
-	// Otherwise fallback to hardcoded default
-	return DefaultUseXattrs
+
+}
+
+// Should tests try to drop GSI indexes before flushing buckets?
+// See SG #3422
+func TestsShouldDropIndexes() bool {
+
+	// First check if the SG_TEST_USE_XATTRS env variable is set
+	dropIndexes := os.Getenv(TestEnvSyncGatewayDropIndexes)
+	switch {
+	case strings.ToLower(dropIndexes) == strings.ToLower(TestEnvSyncGatewayTrue):
+		return true
+	default:
+		// Otherwise fallback to hardcoded default
+		return DefaultDropIndexes
+	}
```

Comment:
Same as above - is there a reason this should be a switch?

Reply:
Fixed